magento/magento2
# Add Tracking URL to UI & API for use email & customer account pages

FAO @vrann 
### Description
Add a new DB field `sales_shipment_track.track_url` which can be input via admin UI on the Create Shipment / Add Track page or can be input via API.
Display this new field if on admin UI, View Shipment page and on Customer UI view account / My Orders / Shipments

### Manual testing scenarios
1. Login as a merchant, view an existing order, click Ship - add tracking URL as part of the tracking information.  (Add two tracks - one as a valid URL as per Zend_Uri validator and another as invalid)
2. Pass track_url as part of request to Create Shipment via API
3. As administrator, check shipment and click on Tracking URL hyperlink - valid URL should open in new window, invalid URL should be native behaviour.
4. As customer, perform same checks as administrator but via Customer UI.

### Contribution checklist
 - [] All commits are accompanied by meaningful commit messages
 - [X ] All new or changed code is covered with unit/integration tests (if applicable)
